pkg/hubble/exporter/cell.go

```diff
@@ -0,0 +1,161 @@
+	ExportFieldmask:        []string{},
+}
+
+func (def config) Flags(flags *pflag.FlagSet) {
+	flags.String("hubble-flowlogs-config-path", def.FlowlogsConfigFilePath, "Filepath with configuration of hubble flowlogs")
+	flags.String("hubble-export-file-path", def.ExportFilePath, "Filepath to write Hubble events to. By specifying `stdout` the flows are logged instead of written to a rotated file.")
+	flags.Int("hubble-export-file-max-size-mb", def.ExportFileMaxSizeMB, "Size in MB at which to rotate Hubble export file.")
+	flags.Int("hubble-export-file-max-backups", def.ExportFileMaxBackups, "Number of rotated Hubble export files to keep.")
+	flags.Bool("hubble-export-file-compress", def.ExportFileCompress, "Compress rotated Hubble export files.")
+	flags.StringSlice("hubble-export-allowlist", []string{}, "Specify allowlist as JSON encoded FlowFilters to Hubble exporter.")
+	flags.StringSlice("hubble-export-denylist", []string{}, "Specify denylist as JSON encoded FlowFilters to Hubble exporter.")
+	flags.StringSlice("hubble-export-fieldmask", def.ExportFieldmask, "Specify list of fields to use for field mask in Hubble exporter.")
+}
+
+func (cfg config) validate() error {
+	if fm := cfg.ExportFieldmask; len(fm) > 0 {
+		_, err := fieldmaskpb.New(&flowpb.Flow{}, fm...)
+		if err != nil {
+			return fmt.Errorf("hubble-export-fieldmask contains invalid fieldmask '%v': %w", fm, err)
+		}
+	}
+	return nil
+}
+
+type hubbleExportersOut struct {
+	cell.Out
+
+	// use flatten as a poor man's optional field.
+	ObserverOptions []observeroption.Option `group:"hubble-observer-options,flatten"`
+}
+
+type hubbleExportersParams struct {
+	cell.In
+
+	Lifecycle cell.Lifecycle
+	Config    config
+
+	OnExportEvent []exporteroption.OnExportEvent `group:"hubble-exporter-on-export-event"`
+
+	// TODO: replace by slog
+	Logger logrus.FieldLogger
+}
+
+func newHubbleStaticExporter(params hubbleExportersParams) (hubbleExportersOut, error) {
+	if err := params.Config.validate(); err != nil {
+		return hubbleExportersOut{}, fmt.Errorf("hubble-exporter configuration error: %w", err)
+	}
+
+	if params.Config.ExportFilePath != "" {
+		exporterOpts := []exporteroption.Option{
+			exporteroption.WithPath(params.Config.ExportFilePath),
+			exporteroption.WithMaxSizeMB(params.Config.ExportFileMaxSizeMB),
+			exporteroption.WithMaxBackups(params.Config.ExportFileMaxBackups),
+			exporteroption.WithAllowList(params.Logger, params.Config.ExportAllowlist),
+			exporteroption.WithDenyList(params.Logger, params.Config.ExportDenylist),
+			exporteroption.WithFieldMask(params.Config.ExportFieldmask),
+		}
+		if params.Config.ExportFileCompress {
+			exporterOpts = append(exporterOpts, exporteroption.WithCompress())
+		}
+		for _, onExportEvent := range params.OnExportEvent {
+			exporterOpts = append(exporterOpts, exporteroption.WithOnExportEvent(onExportEvent))
+		}
+		exporter, err := NewExporter(params.Logger, exporterOpts...)
+		if err == nil {
```

Comment:
again here shortcuting would help readers IMO
```go
		if err != nil {
			params.Logger.WithError(err).Error("Failed to configure Hubble static exporter")
			return hubbleExportersOut{}, nil
		}
		…
```
Also question: why don't we return the error to the caller?

Reply:
The current logic is non-failing at the moment and I wasn't sure if we wanted to change that behaviour in this PR.
But happy to do so if warrant this a fatal failure for the daemon.

Reply:
Is it possible for our cell to be marked non-critical so we can just return errors as normal, but not cause the entire agent to fail? Ideally this would be surfaced so that the "hubble flow export cell is unhealthy", which I know we have some mechanisms for, but I'm not sure exactly what they are.

Reply:
It seems possible to do something like that using job.OneShot for which the callback is passed a cell.Health we could use for that purpose, will look into it.

Reply:
By moving to job.OneShot for the dynamic exporter, we automatically get a health status attached, and it becomes degraded if an error is returned.

For the static exporter, since its hook-based, I didn't feel like it would make sense to declare a OneShot job only to declare a status (which is not logged btw and is only viewable using cilium-dbg).